dvc/lock.py

```diff
@@ -58,7 +58,7 @@ def __init__(self, lockfile, tmp_dir=None):
             # [1] https://github.com/iterative/dvc/issues/2582
             self._hostname = socket.gethostname()
 
-            self._lockfile = lockfile
+            self._lockfile = fspath(lockfile)
```

Comment:
Feels like pathlib&fspath parts of this PR only dilute it, do we really need it? I'm afraid of unexpected issues happening in the code that we affect by these.

Reply:
we don't _need it_, we can work with strings as paths, but I thought the idea was to push towards working with _pathlike_ objects.

Do you want me to remove those parts and include them in a separate PR, @efiop ?
I understand the feeling, but that's why we have tests, right? To bring some sense of confidence. :slightly_smiling_face: 

Those are not arbitrary changes on `fspath`, it is currently not working for `pathlib.Path` in python 3.5 because it lacks `__fspath__` implementation.
The change address this in 2 lines of code with a comment about why it was added.

---

What direction are we going to take? Using `PathInfo` everywhere (ignoring support for `pathlib.Path`), a combination of _strings_ and `PathInfo` to represent paths or something else?

Reply:
@mroutis I'm really sorry for adding some work on top, but yes, let's move pathlib to another issue/pr. Feels like it obstructs the view for this one.

Reply:
It's all right, @efiop :slightly_smiling_face: 
I'll move to another PR.